pkg/helpers/sa_syncer.go

```diff
@@ -0,0 +1,92 @@
+package helpers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+// EnsureSAToken get the saToken of target sa and ensure it was rendered as expected.
+// A usage example combined with RenderToKubeconfigSecert would be like the following:
+// ...
+// saName := clusterManagerName+"-registration-controller-sa"
+// secretName := saName+"-kubeconfig"
+// err := helpers.EnsureSAToken(ctx, saName, clustermanagerNamespace, hostedAmdinKubeClient,
+// 	helpers.RenderToKubeconfigSecret(secretName, clustermanagerNamespace, hostedAdminKubeconfig, kubeClient, recorder)) // kubeClient used to create secret; normally recorder is from controller object.
+// if err != nil {
+// 	return err
+// }
+// ...
+func EnsureSAToken(ctx context.Context, saName, saNamespace string, hostedClient kubernetes.Interface, renderSAToken func([]byte) error) error {
+	// get the service account
+	sa, err := hostedClient.CoreV1().ServiceAccounts(saNamespace).Get(ctx, saName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if len(sa.Secrets) == 0 {
+		return fmt.Errorf("token secret for %s not exist yet", saName)
+	}
+	tokenSecretName := sa.Secrets[0].Name
```

Comment:
do we need to check the name of secret in sa, like xxx-token-xx?

Reply:
I thought the token secret would be the first secret as default, I don't know whether it was promised or not. It's based on observation. 
However, I'm also not sure about the secret name. Does the format of the name **have to** be "xxx-token-xx"? Is there any case that the token secret's name doesn't contain a "token"?

Reply:
Done. Now if no secret's name contains "token", the function would return an error.